src/daemon/Display.cpp

```diff
@@ -391,13 +433,14 @@ namespace SDDM {
     }
 
     void Display::slotHelperFinished(Auth::HelperExitStatus status) {
-        // Don't restart greeter and display server unless sddm-helper exited
-        // with an internal error or the user session finished successfully,
-        // we want to avoid greeter from restarting when an authentication
-        // error happens (in this case we want to show the message from the
-        // greeter
-        if (status != Auth::HELPER_AUTH_ERROR)
+        // Restart the greeter and display server if sddm-helper exited
+        // with an internal error or a session error, otherwise we just
+        // want to reuse the existing so we jump to its VT
+        if (status == Auth::HELPER_SESSION_ERROR ||
+                status == Auth::HELPER_OTHER_ERROR)
             stop();
+        else
```

Comment:
If I've gone through the greeter, quit that, spawned a user session and then logged out gracefully, we need to restart the greeter.

This else statement prevents that, we just VT switch to a somewhat dead session.

Reply:
With https://github.com/plfiorini/sddm/pull/5 I don't see the problem anymore. I'm not sure how to reproduce the problem you're seeing here.

Reply:
I'm confused how it works then.

I log out, my user session helper stops, I switch to this VT, who restarts the greeter?

Reply:
AHA! I think I know why it works for you.

When we log in, the greeter stays running.
On the default theme when we go back to the greeter and try to log in again, this is fine.

On the breeze theme because we fade away and disable things. We do switch back to the greeter but it's in a broken state.

Is it the design that the greeter stay running? 

Also is m_terminalId correct here in the cases that the greeter helper spawns X?

Reply:
> Is it the design that the greeter stay running?

yes, one of the patches makes the greeter stay alive hence we just need to switch vt

Reply:
>yes, one of the patches makes the greeter stay alive hence we just need to switch vt

It's quite a bit of memory to waste keeping it and Xorg alive, just to make it slightly faster if someone does log out. I don't think it's a good trade off.

Also as observed it introduces a compatibility issue.

Reply:
Now the user sessions is closed and we see a prompt and then the greeter is shown.
Perhaps we can have this happening:

1. helper tells daemon that the session is about to quit
2. daemon spawns the greeter and switch to its vt, then tells the helper about it
3. helper quits

But we shouldn't be worrying about this stuff in this PR.
This has nothing to do with running xorg with the sddm user.